libmamba/tests/src/download/test_mirror.cpp

```diff
@@ -58,7 +58,8 @@ namespace mamba::download
         TEST_CASE("PassThroughMirror")
         {
             std::unique_ptr<Mirror> mir = make_mirror("");
-            CHECK_EQ(typeid(*mir), typeid(PassThroughMirror));
+            auto& ref = *mir;
```

Comment:
How about the warning for this change? I believe this is more related to performance but we only dereference once.

Reply:
This is not about performance, compiler didn't like that there possibly was a call with side effects in `typeid` call (dereferencing might fail).
The easiest way to fix this is to dereference in a separate line.

Reply:
Well in that case, it still doesn't solve the problem of dereferencing possible failure.
I suggest adding instead a `static_assert` before the `CHECK_EQ`:
`static_assert(mir != nullptr);`

Reply:
I can't do it, because `mir` is not known in a compile-time, so `static_assert` won't work here

Reply:
And maybe also rename `ref` to something more explicit, `mir_ref` for example.

Reply:
Done